docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+options.
+
+The following figure shows the key master and agent instances on separate machines:
+
+.. image:: /assets/images/arch00.png
+
+However, a single machine can have both a master and one agent instance.
+
+Master and Agent Nodes
+======================
+
+Determined comprises a single *master* and one or more *agents* in a cluster environment. The master
+is a single, non-GPU node that manages the cluster.
+
+The master keeps experiment metadata in a PostgreSQL database, which can be queried using the WebUI
+or CLI. You can interact with the master, directly, but cannot interact with agents.
+
+Master Node Scope of Responsibility
+-----------------------------------
+
+Master node responsibilities:
+
+-  Stores experiment, trial, and workload metadata.
+-  Schedules and dispatch experiments to agents.
+-  Dynamically provisions and deprovisions on-premises and cloud agents.
+-  Advances experiment, trial, and workload state machines.
+-  Responds to CLI commands.
+-  Hosts the WebUI, which is primarily used to monitor experiments.
+-  Serves the REST API.
+
+The master is also responsible for the following training-specific searcher, metrics, checkpointing,
+and scheduling functionality:
+
++---------------+----------------------------------------------------------------------+
+| Function      | Description                                                          |
++===============+======================================================================+
+| Searcher      | The *Searcher* implements a hyperparameter search algorithm and is   |
+|               | responsible for coordinating the work of all experiment trials.      |
++---------------+----------------------------------------------------------------------+
+| Metrics       | *Metrics* provides persistent storage of metrics reported by Trials. |
++---------------+----------------------------------------------------------------------+
+| Checkpointing | *Checkpointing* captures the training state at a given time in the   |
+|               | *model registry*.                                                    |
++---------------+----------------------------------------------------------------------+
+| Scheduling    | *Scheduling* schedules jobs to run, ensuring that all of the compute |
+|               | resources required for a job are available before the job launches.  |
++---------------+----------------------------------------------------------------------+
+
+Agent Node Scope of Responsibility
+----------------------------------
+
+Agent node responsibilities:
+
+-  Discovers local computing devices and sends device/slot metadata to the master.
+-  Runs workloads at the request of the master.
+-  Monitors containers and sends container information to the master.
+-  For a *trial runner*, which runs a trial in a containerized environment, reports trial runner
+   states to the master.
```

Comment:
Let's delete this trial runner bullet point.  What it is saying is basically a subset of what the previous bullet said.